agent/connect/ca/provider_vault.go

```diff
@@ -437,7 +438,7 @@ func (v *VaultProvider) Sign(csr *x509.CertificateRequest) (string, error) {
 		return "", fmt.Errorf("issuing_ca was not a string")
 	}
 
-	return fmt.Sprintf("%s\n%s", cert, ca), nil
+	return fmt.Sprintf("%s%s", AddSingleNewline(cert), AddSingleNewline(ca)), nil
```

Comment:
non blocking nit: `return AddSingleNewline(cert) + AddSingleNewline(ca)` might read a little better and would match how it's done in the `xds` package (using `+` for concat).